### PR TITLE
[MTDSA-4316] uk property models

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      | "totalDeductions" : 60.00,
+      | "netProfit": 20.00,
+      | "taxableProfit" : 10.00,
+      | "netLoss": 10.00,
+      | "taxableLoss" : 35.00,
+      | "accountingAdjustments": -30.00
+      |}
+    """.stripMargin)
+
+  val jsonString: String ="""{"total":{"income":100.00,"expenses":50.00,"additions":5.00,"deductions":60.00},"accountingAdjustments":-30.00,"profit":{"net":20.00,"taxable":10.00},"loss":{"net":10.00,"taxable":35.00}}"""
```
Comment (BetaDraconis) [inline test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala]: Can you give this proper JSON formatting (i.e like the desJson value above)

File: test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  val model =
+    RetrieveUKPropertyBISSResponse (
```
Comment (BetaDraconis) [inline test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala]: Add type annotation.

File: test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala (hunk 1 above)
Comment (james-work-account) [reply test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala]: `model.toJsonString` outputs:
```
{"total":{"income":100.00,"expenses":50.00,"additions":5.00,"deductions":60.00},"accountingAdjustments":-30.00,"profit":{"net":20.00,"taxable":10.00},"loss":{"net":10.00,"taxable":35.00}}
```
Formatting the `jsonString` variable will just mean that in the comparison part of the tests you need to remove all excess whitespace, which is unnecessary imo.

File: test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala (hunk 2 above)
Comment (james-work-account) [reply test/v1/models/response/RetrieveUKPropertyBISSResponseSpec.scala]: Are type annotations necessary when the type is so obvious?

File: app/v1/models/response/package.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package v1.models
+
+import org.json4s.CustomSerializer
+import org.json4s.JsonAST.JDecimal
+
+package object response {
+  object BigDecimalSerializer extends CustomSerializer[BigDecimal](format =>
```
Comment (BetaDraconis) [inline app/v1/models/response/package.scala]: The 'format' parameter doesn't seem to be used.

**Comment by Nboaram** [comment]
Do not merge this.

**Comment by MarkAKelly** [comment]
I think as part of the branch restoration, some of the original code has been lost (I'm looking at you BigDecimalSerializer).

@james-work-account do you mind taking a look at this when you have a chance and double checking that all of the bits you had intended to have in here are still there?

File: app/v1/models/response/package.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package v1.models
+
+import org.json4s.CustomSerializer
+import org.json4s.JsonAST.JDecimal
+
+package object response {
+  object BigDecimalSerializer extends CustomSerializer[BigDecimal](_ =>
```
Comment (james-work-account) [inline app/v1/models/response/package.scala]: @MarkAKelly BigDecimalSerializer was moved to a package object. It's still here.

**Pull request merged**